src/GridFieldNestedForm.php

```diff
@@ -0,0 +1,435 @@
+<?php
+
+namespace Symbiote\GridFieldExtensions;
+
+use Exception;
+use SilverStripe\Admin\ModelAdmin;
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\Director;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Control\HTTPResponse_Exception;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Forms\GridField\AbstractGridFieldComponent;
+use SilverStripe\Forms\GridField\GridField;
+use SilverStripe\Forms\GridField\GridField_ColumnProvider;
+use SilverStripe\Forms\GridField\GridField_DataManipulator;
+use SilverStripe\Forms\GridField\GridField_HTMLProvider;
+use SilverStripe\Forms\GridField\GridField_SaveHandler;
+use SilverStripe\Forms\GridField\GridField_URLHandler;
+use SilverStripe\Forms\GridField\GridFieldDetailForm_ItemRequest;
+use SilverStripe\Forms\GridField\GridFieldStateAware;
+use SilverStripe\ORM\DataList;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\DataObjectInterface;
+use SilverStripe\ORM\Hierarchy\Hierarchy;
+use SilverStripe\ORM\SS_List;
+use SilverStripe\Versioned\Versioned;
+use SilverStripe\View\ViewableData;
+use Symbiote\GridFieldExtensions\GridFieldOrderableRows;
+
+/**
+ * Gridfield component for nesting GridFields
+ */
+class GridFieldNestedForm extends AbstractGridFieldComponent implements
+    GridField_URLHandler,
+    GridField_ColumnProvider,
+    GridField_SaveHandler,
+    GridField_HTMLProvider,
+    GridField_DataManipulator
+{
+    use Configurable, GridFieldStateAware;
+    
+    const POST_KEY = 'GridFieldNestedForm';
+
+    private static $allowed_actions = [
+        'handleNestedItem'
+    ];
+
+    private static $max_nesting_level = 10;
+    
+    /**
+     * @var string
+     */
+    protected $name;
+    protected $expandNested = false;
+    protected $forceCloseNested = false;
+    protected $gridField = null;
+    protected $record = null;
+    protected $relationName = 'Children';
+    protected $inlineEditable = false;
+    protected $canExpandCheck = null;
+    protected $maxNestingLevel = null;
+    
+    public function __construct($name = 'NestedForm')
+    {
+        $this->name = $name;
+    }
+    
+    /**
+     * Get the grid field that this component is attached to
+     * @return GridField
+     */
+    public function getGridField()
+    {
+        return $this->gridField;
```

Comment:
```suggestion
    public function getGridField(): GridField
    {
        return $this->gridField;
```
Please add strict typehints for arguments and return types for all methods in this PR which aren't inherited from superclasses or interfaces.

Note that PHPDocs are only required for parameters and return types if there is additional information, such as a description. If it's just the type, we don't need the PHPDoc when there's a real typehint.